### PR TITLE
fix: MultiActionButton and SplitButton tabbing issues - FE-4322

### LINKED ISSUE #4352: MultiActionButton and SplitButton backtabbing is not working as expected
### Current behaviour

1) Not able to move to parent, sub-menu closed after pressing shift + tab but no focus on any elements

### Expected behaviour

1) shift+tab puts a focus on the parent for MultiActionButton and SplitButton

### CodeSandbox or Storybook URL

https://codesandbox.io/s/loading-spinner-for-export-forked-xjxml?file=/src/index.js

### Suggested Solution

_No response_

### Carbon Version

77.14.1

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS, Windows

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour

Fixes #4352 
Pressing Tab while SplitButton or MultiActionButton is opened will focus next focusable element on page
Pressing Shift + Tab while SplitButton or MultiActionButton is opened will focus previous focusable element on page
### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

https://codesandbox.io/s/loading-spinner-for-export-forked-rfxqi

### CODE REVIEW

File: cypress/integration/common/multiActionButton.feature [hunk 1]
```diff
@@ -48,4 +48,12 @@ Feature: Multi Action Button default component
...
-    Then Multi Action Button in a hidden container is expanded and contains three items
\ No newline at end of file
+    Then Multi Action Button in a hidden container is expanded and contains three items
+
+  @positive
+  Scenario: I press tab key while Multi Action Button is open
+    Given I open "Multi Action Button" component page "button types"
+      And I click multi action button toggle
+      And I focus second additional button
+      And I press tab key on Multi Action Button
```
Comment (DlgSHi) [inline cypress/integration/common/multiActionButton.feature:58]:
```suggestion
      When I press tab key on Multi Action Button
```

File: cypress/integration/common/splitButton.feature [hunk 2]
```diff
@@ -77,4 +77,12 @@ Feature: Split Button component
...
-    Then Split Button is expanded
\ No newline at end of file
+    Then Split Button is expanded
+
+  @positive
+  Scenario: I press tab key while Split Button is open
+    Given I open "Split Button" component page "button types"
+      And I click split button toggle
+      And I focus second additional button
+      And I press tab key on split button
```
Comment (DlgSHi) [inline cypress/integration/common/splitButton.feature:87]:
```suggestion
      When I press tab key on split button
```

**Review by DlgSHi — APPROVED** [review]
Approve for a `cypress` change, expect the failing test :)

File: cypress/integration/common/splitButton.feature [hunk 3]
```diff
@@ -77,4 +77,12 @@ Feature: Split Button component
...
-    Then Split Button is expanded
\ No newline at end of file
+    Then Split Button is expanded
+
+  @positive
+  Scenario: I press tab key while Split Button is open
+    Given I open "Split Button" component page "button types"
+      And I click split button toggle
+      And I focus second additional button
+    When Start to type your When step here I press tab key on split button
```
Comment (DlgSHi) [inline cypress/integration/common/splitButton.feature:87]:
```suggestion
When I press tab key on split button
```

**Review by DipperTheDan — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Comment by harpalsingh** [comment]
Need to look at some of the styles on focus but the functionality is fine, as per Martins comments we we look further into other improvements where we will use enter to open children menus and will impact several components but need to design/test that first.

**Review by edleeks87 — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

File: cypress/integration/common/splitButton.feature [hunk 4]
```diff
@@ -77,4 +77,12 @@ Feature: Split Button component
...
\ No newline at end of file
+    Then Split Button is expanded
+
+  @positive
+  Scenario: I press tab key while Split Button is open
+    Given I open "Split Button" component page "button types"
+      And I click split button toggle
+      And I focus second additional button
+    When I press tab key on split button
+    Then Second Split Button component is focused
```
Comment (ZhuoyuJin) [inline cypress/integration/common/splitButton.feature:88]: Then The second Split Button is focused

File: cypress/support/step-definitions/split-button-steps.js [hunk 5]
```diff
@@ -106,6 +107,22 @@ When("I hover mouse onto split button", () => {
...
+
+When("I press tab key on split button", () => {
+  splitToggleButton().eq(0).tab();
+});
+
+When("I focus second additional button", () => {
+  additionalButton(1).focus();
+});
+
+Then("Second Split Button component is focused", () => {
```
Comment (ZhuoyuJin) [inline cypress/support/step-definitions/split-button-steps.js:122]: Then("The second Split Button is focused", () => {

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**